### PR TITLE
SALTO-1382 added core annotation '_service_id' to check if type is service_id type

### PR DESCRIPTION
**Pull request by omrifarkash**
_Release Notes_: 
Standard built in type "SERVICE_ID" has a new annotation '_service_id' = true
We use this annotation to decide wether a type is a SERVICE_ID or not.

### CODE REVIEW

**Comment by omrifarkash** [comment]
@adamro @ori-moisis 
ping
when you have time

File: packages/salesforce-adapter/src/filters/utils.ts [hunk 1]
```diff
@@ -88,12 +88,12 @@ void => {
     log.debug(`added API_NAME=${fullApiName} to ${elem.elemID.name}`)
   }
   if (!isField(elem) && !elem.annotationRefTypes[API_NAME]) {
-    elem.annotationRefTypes[API_NAME] = createRefToElmWithValue(BuiltinTypes.SERVICE_ID)
+    elem.annotationRefTypes[API_NAME] = createRefToElmWithValue(Types.primitiveDataTypes.ServiceId)
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/utils.ts:91]: Unrelated to this PR

File: packages/salesforce-adapter/src/filters/utils.ts [hunk 2]
```diff
@@ -88,12 +88,12 @@ void => {
...
   }
 }
 
 export const addMetadataType = (elem: ObjectType, metadataTypeValue = CUSTOM_OBJECT): void => {
-  setAnnotationDefault(elem, METADATA_TYPE, metadataTypeValue, BuiltinTypes.SERVICE_ID)
+  setAnnotationDefault(elem, METADATA_TYPE, metadataTypeValue, Types.primitiveDataTypes.ServiceId)
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/utils.ts:96]: Unrelated to this PR

File: packages/salesforce-adapter/src/filters/workflow.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ import { FilterCreator } from '../filter'
 import {
   apiName, metadataType, createInstanceElement, metadataAnnotationTypes, MetadataTypeAnnotations,
   toMetadataInfo,
+  Types,
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/workflow.ts:36]: nit: If we keep adding lines like this I prefer the

```
import { apiName, metadataType, createInstanceElement, metadataAnnotationTypes, MetadataTypeAnnotations, toMetadataInfo, Types }
```

style but it's personal pref

File: packages/salesforce-adapter/src/filters/workflow.ts [hunk 4]
```diff
@@ -125,7 +126,9 @@ const createDummyWorkflowInstance = async (
   const workflowType = new ObjectType({
     elemID: new ElemID(SALESFORCE, WORKFLOW_METADATA_TYPE),
     fields: {
-      [INSTANCE_FULL_NAME_FIELD]: { refType: createRefToElmWithValue(BuiltinTypes.SERVICE_ID) },
+      [INSTANCE_FULL_NAME_FIELD]: {
+        refType: createRefToElmWithValue(Types.primitiveDataTypes.ServiceId),
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/workflow.ts:130]: Both unrelated to this PR and not needed

**Review by adamro — COMMENTED** [review]
I stopped in the middle cause I am a bit confused.

I thought we said we are separating this to two PR's - 
1. Change the core behaviour to check if something is a serviceId by annotation and not ElemID
2. Do the Salesforce implementation of CREATEABLE/UPDATEABLE/QUERYABLE based on 1 and using a Salesfroce specific ServiceId

It seems like this PR now includes both 1 and 2 whilst I was expecting it to only be 1 and see a separate PR for 2. So a lot of my comments are already about things not being related to this PR (cause I thought it was only 1).
For 2, if this was the implementations, I have more logical comments but I wouldn't want them to be here.

Bottom line - Split this to two PR's please and ping me to re-review 1 and then 2 separately please.
